baselines/fedvssl/README.md

```diff
@@ -0,0 +1,168 @@
+
+## Experimental Setup
+
+**Task:** Action Recognition
+
+**Model:** 
+* This directory first pretrain Catch-the-Patch (CtP) model from the CtP (see `CtP/pyvrl/models/pretraining/ctp`) repository during FL pretrainin stage The backbone model is R3D-18 (see `/CtP/pyvrl/models/backbones/r3d.py`). 
+* After pretraining it finetunes the R3D-18 model on UCF-101 dataset.
+
+**Dataset:** This baselines only include pre-training and fine-tuning with UCF-101 dataset. However, we also provide the script files to generate the partitions for Kinetics-400 datasets. 
+For UCF-101 dataset, one can simply run the `dataset_preparation.py` file to download and generate the iid splits for UCF-101 datasets.
+
+| Dataset | #classes | #partitions | partitioning method |  partition settings  |
+|:--------|:--------:|:-----------:| :---: |:--------------------:|
+| UCF101  |   101    |     10      | randomly partitioned |       uniform        |
+| Kinectics-400    |   400    |     100     | randomly partitioned | 8 classes per client |
+
+**Training Hyperparameters:** The following table shows the main hyperparameters for this baseline with their default value (i.e. the value used if you run python main.py directly)
+
+| Description        |            Default Value            |
+|:-------------------|:-----------------------------------:|
+| total clients      |                 2                  |
+| clients per round  |                 2                  | 
+| number of rounds	  |                 20                  | 
+| client resources	  | {'num_cpus': 2.0, 'num_gpus': 1.0 } | 
+| optimizer	  |                 SGD                  | 
+| alpha coefficient	 |                 0.9                 | 
+| beta coefficient	  |                  1                  | 
+
+## Environment Setup
+Please make sure you have installed CUDA 11.0 on your machine 
+(see [NVIDIA CUDA Toolkit website](https://developer.nvidia.com/cuda-11.0-download-archive?target_os=Linux)).
+To construct the Python environment follow these steps:
+
+```bash
+# install the base Poetry environment
+poetry install
+
+# activate the environment
+poetry shell
+
+# install PyTorch with GPU support.
+pip install torch==1.7.1+cu110 torchvision==0.8.2+cu110 torchaudio==0.7.2 -f https://download.pytorch.org/whl/torch_stable.html
+
+# install mmcv package
+pip install mmcv==1.2.4
+```
+
+## Running the Experiments
+To run FedVSSL with UCF-101 baseline, please ensure you have activated your Poetry environment (execute `poetry shell` from this directory). Then, download the `CtP` repo, download the datasets and preprocess it:
+
+```bash
+# clone CtP repo
+git clone https://github.com/yan-gao-GY/CtP.git fedvssl/CtP
+
+sudo apt install unrar
+
+cd fedvssl
+python dataset_preparation.py
+
+cd ..
+```
+
+
+Finally, we can launch the training:
+
+```bash
+# run federated SSL training with FedVSSL
+python -m fedvssl.main pre_training=true # this will run using the default settings.
```

Comment:
if readers are presented with this as the first command, I think it would be better if first it is mentioend that they need to download one of the checkpoints, and either update the path to it in the `conf/base.yaml` or pass it as:
```bash
python -m fedvssl.main pre_training=true pretrained_model_path=<CHECKPOINT>.npz
```

Reply:
The checkpoints are used for downstream fine-tuning. The command here is to run SSL pre-training on UCF-101 dataset from scratch.

To run fine-tuning:
`python -m fedvssl.main pre_training=false pretrained_model_path=<CHECKPOINT>.npz`
The `pretrained_model_path` can be set to the checkpoints path trained by users on UCF-101 from scratch, or the downloaded ones which were trained on K-100 dataset.